src/map.cpp

```diff
@@ -1430,33 +1446,44 @@ void Map::timerUpdate(float dtime, float unload_timeout,
 			MapBlock *block = (*i);
 
 			block->incrementUsageTimer(dtime);
+			mapblock_queue.push(TimeOrderedMapBlock(sector, block));
```

Comment:
I think it would be better to only push blocks once, when they are created, in a global mapblock queue, and not on every step. However, it would have the drawback that already-pushed mapblocks would need to be kept in the same order, that would probably imply using the current order; so it you want to block keeping criterion to be changed later, it would not be a good idea.